src/THORSwap/TWSwap.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright © 2017-2021 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include <TrustWalletCore/TWTHORSwap.h>
+#include "Swap.h"
+#include "proto/THORSwap.pb.h"
+#include "Data.h"
+
+using namespace TW::THORSwap;
+using namespace TW;
+
+TWData *_Nonnull TWTHORSwapSwap(TWData *_Nonnull input) {
+    Proto::SwapInput inputProto;
+    Proto::SwapOutput outputProto;
+    if (!inputProto.ParseFromArray(TWDataBytes(input), static_cast<int>(TWDataSize(input)))) {
+        // error
+        outputProto.set_error("Could not deserialize input proto");
```

Comment:
Did we have list of all possible errors? maybe move to const somewhere to reuse?

Reply:
Reworked error handling, errors defined in proto.